Scripts/Editor/Debug/ScriptDefineDefinition.cs

```diff
@@ -0,0 +1,50 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Anvil.Unity.Editor.Debug
+{
+    /// <summary>
+    /// Defines information for use with a Script Define and the behaviour when toggling on and off.
+    /// <see cref="ScriptDefinesToggle"/>
+    /// </summary>
+    public class ScriptDefineDefinition
+    {
+        internal string Define { get; }
+        internal string MenuPath { get; }
+        internal HashSet<string> RequiredDefines { get; }
+        internal HashSet<string> DependentDefines { get; }
```

Comment:
We talked about this in another PR in the DOTS repo and auto-getters are the same as readonly fields. 

I have taken to using the pattern of making anything `public`, `protected` or `internal` into auto-getters because then it's easier to use for interfaces in the future.